pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -114,6 +116,12 @@ func (s *ServiceEntryStore) workloadEntryHandler(old, curr config.Config, event
 		namespace: curr.Namespace,
 	}
 
+	// If an entry is unhealthy, we will mark this as a delete instead
+	// This ensures we do not track unhealthy endpoints
+	if !isHealthy(curr) {
```

Comment:
Also, should this be guarded with the feature flag? If a user restarts pilot to disable health-checking, I think the expectation is that even "unhealthy" entries are used in EDS.

Reply:
Good point, I forgot we have a feature flag on istiod as well. I will add it, we can probably remove it in the next release - this is purely a safety knob not a real user configuration I think (if you don't want health checks, don't configure them)